ddtrace/contrib/internal/ray/patch.py

```diff
@@ -0,0 +1,488 @@
+            raise e
+
+
+def traced_wait(wrapped, instance, args, kwargs):
+    """
+    Trace the calls of ray.wait
+    """
+    if not tracer:
+        return wrapped(*args, **kwargs)
+
+    if tracer.current_span() is None:
+        tracer.context_provider.activate(_extract_tracing_context_from_env())
+
+    with long_running_ray_span(
+        "ray.wait",
+        service=os.environ.get("_RAY_SUBMISSION_ID"),
+        span_type=SpanTypes.RAY,
+        child_of=tracer.context_provider.active(),
+        activate=True,
+    ) as span:
+        span.set_tag_str(SPAN_KIND, SpanKind.PRODUCER)
+        timeout = kwargs.get("timeout")
+        num_returns = kwargs.get("num_returns")
+        fetch_local = kwargs.get("fetch_local")
+        if timeout is not None:
+            span.set_tag_str("ray.wait.timeout_s", str(timeout))
+        if num_returns is not None:
+            span.set_tag_str("ray.wait.num_returns", str(num_returns))
+        if fetch_local is not None:
+            span.set_tag_str("ray.wait.fetch_local", str(fetch_local))
+        _inject_ray_span_tags(span)
+        return wrapped(*args, **kwargs)
+
+
+def _job_supervisor_run_wrapper(method: Callable[..., Any]) -> Any:
+    async def _traced_run_method(self: Any, *args: Any, _dd_trace_ctx=None, **kwargs: Any) -> Any:
+        from ddtrace import tracer
+        from ddtrace.ext import SpanTypes
+
+        if not tracer or _dd_trace_ctx is None:
+            return await method(self, *args, **kwargs)
+
+        context = _TraceContext._extract(_dd_trace_ctx)
+        submission_id = os.environ.get(RAY_SUBMISSION_ID)
+
+        with long_running_ray_span(
+            f"{self.__class__.__name__}.{method.__name__}",
+            service=submission_id,
+            span_type=SpanTypes.RAY,
+            child_of=context,
+            activate=True,
+        ) as supervisor_run_span:
+            _inject_context_in_env(supervisor_run_span.context)
+
+            try:
+                await method(self, *args, **kwargs)
+            except ray.exceptions.AsyncioActorExit as e:
+                # if the job succeeded we remove from the span
+                # the error used to exit the actor
+                job_info = await self._job_info_client.get_info(submission_id)
+
+                if str(job_info.status) == "FAILED":
+                    raise e
+
+    return _traced_run_method
+
+
+def _exec_entrypoint_wrapper(method: Callable[..., Any]) -> Any:
+    def _traced_run_method(self: Any, *args: Any, _dd_trace_ctx=None, **kwargs: Any) -> Any:
+        from ddtrace import tracer
+        from ddtrace.ext import SpanTypes
+
+        if not tracer:
+            return method(self, *args)
+
+        if tracer.current_span() is None:
+            tracer.context_provider.activate(_extract_tracing_context_from_env())
+
+        # Without this the name of the span will contain a path which will break tests
+        if os.environ.get("_DD_TRACE_RAY_TESTING"):
```

Comment:
Is this for snapshot tests? If so we can use snapshot(ignores=...). This will allow you to skip validation on certain fields. I am not a big fan of testing specific branches but it's fine if there are no good workarounds

Reply:
I'm aware about ignores, the problem is that if we ignore we cannot properly test the span. If you feel it is ok to not test it, I can remove it